tsconfig.json

```diff
@@ -0,0 +1,19 @@
+{
+  "compilerOptions": {
+    "lib": ["es2020"],
+    "moduleResolution": "nodenext",
+    "module": "esnext",
+    "resolveJsonModule": true,
+    "allowJs": true,
+    "checkJs": true,
+    "noEmit": false,
+    "declaration": true,
+    "declarationMap": true,
+    "emitDeclarationOnly": true,
+    "strict": true,
+    "target": "es5",
```

Comment:
Should this be ES6?

Reply:
Although our Node `engine` versions should be safe for ES6, I don't know if you mean to preserve a wider browser compatibility? (To be more out-of-the-box browser ready, [eslint-plugin-ecmascript-compat](https://github.com/robatwilliams/es-compat/blob/master/packages/eslint-plugin-ecmascript-compat/README.md) could help, though I imagine the project may prefer to push that responsibility (and choice of supported environment) downstream.)

If just worrying about Node, if https://node.green/#ES2019 is any indication, we could go as high as `es2019`.

Reply:
I think ES6 is well enough supported in browsers and Node.JS that we should target that as the minimum.

Reply:
I've added such a commit. We technically don't need any target here since we're not compiling JavaScript out of TypeScript, but I figured it might be good to settle on a target in case we do and wish to share the same basic config across projects.